src/plugins/chart_expressions/expression_metric/public/components/metric_vis.tsx

```diff
@@ -219,7 +232,33 @@ export const MetricVis = ({
       return nonNumericMetric;
     }
 
-    const baseMetric: MetricWNumber = {
+    if (isArrayOfStrings(value)) {
+      const formattedValues = value.map((v) => formatPrimaryMetric(v));
+      const nonNumericArrayMetric: MetricWStringArrayValues = {
+        value: formattedValues,
+        title: String(title),
+        subtitle,
+        icon: config.metric?.icon ? getIcon(config.metric?.icon) : undefined,
+        extra: renderSecondaryMetric(data.columns, row, config),
+        color: config.metric.color ?? defaultColor,
+      };
+      return nonNumericArrayMetric;
+    }
+
+    if (isArrayOfNumbers(value)) {
+      const numericArrayMetric: MetricWNumberArrayValues = {
+        value,
+        valueFormatter: formatPrimaryMetric,
+        title: String(title),
+        subtitle,
+        icon: config.metric?.icon ? getIcon(config.metric?.icon) : undefined,
+        extra: renderSecondaryMetric(data.columns, row, config),
+        color: config.metric.color ?? defaultColor,
+      };
+      return numericArrayMetric;
+    }
+
+    const baseMetric: MetricWNumber | MetricWNumberArrayValues = {
```

Comment:
The condition above with early return will prevent the metric to be `MetricWNumberArrayValues` at this point.

Reply:
Good catch, fixed in 90f6b947641a9e5db80312a6a5a0195aaaccbf19